util/src/lib.rs

```diff
@@ -1 +1,27 @@
+use tokio::signal;
 
+/// Returns a Future that completes once a signal to shutdown the service is caught.
+pub async fn shutdown_signal() {
+    let ctrl_c = async {
+        signal::ctrl_c() // aka. SIGINT on Unix
+            .await
+            .expect("failed to install Ctrl+C (SIGINT) handler");
+    };
+
+    #[cfg(unix)]
+    let sigterm = async {
+        signal::unix::signal(signal::unix::SignalKind::terminate())
+            .expect("failed to install signal handler for SIGTERM")
+            .recv()
+            .await;
+    };
+    #[cfg(not(unix))]
+    let sigterm = std::future::pending::<()>();
```

Comment:
Do we even need the `cfg`s here? I don't think we will ever want to deploy on Windows?

```suggestion
    let sigterm = async {
        signal::unix::signal(signal::unix::SignalKind::terminate())
            .expect("failed to install signal handler for SIGTERM")
            .recv()
            .await;
    };
```

Reply:
Good point. We shouldn't pretend that the code is portable if we don't intend for it to be. It could in fact be misleading!

Reply:
> I don't think we will ever want to deploy on Windows?

It's possible in the future if other institutes adopt our codebase there might be demand to make it work on Windows. I agree it's not worth considering for now though, we can cross that bridge if and when we come to it, hopefully with some funding for a windows CI runner